src/components/DateInput/DateInput.tsx

```diff
@@ -60,21 +53,137 @@ const parseDate = (str: string, format: string): Date | undefined => {
  * <a href="/#/TextInput">TextInput</a> component (i.e. placeholder, etc.)
  *
  * */
-const DateInput = React.forwardRef<HTMLInputElement, DateInputProps>(function DateInput(
-  { format = 'MM/DD/YYYY', value, onChange, ...rest },
-  ref
-) {
+const DateInput: React.FC<DateInputProps> = ({
+  date,
+  format = 'MM/DD/YYYY',
+  withTime,
+  onChangeDate = noop,
+  ...rest
+}) => {
+  const dateFormatted = date ? dayjs(date) : dayjs();
+  const [currentMonth, setCurrentMonth] = useState(dateFormatted);
+  const [currentDate, setCurrentDate] = useState(date);
+  const [prevDate, setPrevDate] = useState(date);
```

Comment:
Yeap, each time a date is selected is reflected to the input but eventually does not get persisted up until "Apply" button gets clicked. `prevDate` is used in order to restore the previous value upon pressing "Cancel".

Reply:
I think using a `React.ref` might be a better call here, since you don't need a re-render when you do `setPrevDate`. It would also simplify react's dependency arrays, but  I might be missing something

Reply:
`setPrevDate` does not actually affect rendering as the component get rerendered upon closing the dropdown. It's just a placeholder in order to retrieve the previous value.